source/uwp/Renderer/lib/Util.h

```diff
@@ -44,6 +44,34 @@ HRESULT GetSpacingSizeFromSpacing(
     ABI::AdaptiveCards::Uwp::Spacing spacing,
     UINT* spacingSize) noexcept;
 
+HRESULT SetSharedElementProperties(
+    ABI::AdaptiveCards::Uwp::IAdaptiveCardElement * adaptiveCardElement,
+    std::shared_ptr<AdaptiveCards::BaseCardElement> sharedCardElement);
+
+HRESULT GenerateSharedElements(
+    ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveCards::Uwp::IAdaptiveCardElement*>* items,
+    std::vector<std::shared_ptr<AdaptiveCards::BaseCardElement>>& containedElements);
+
+HRESULT GenerateSharedAction(
+    ABI::AdaptiveCards::Uwp::IAdaptiveActionElement* action,
+    std::shared_ptr<AdaptiveCards::BaseActionElement>& sharedAction); 
+
+HRESULT GenerateSharedActions(
+    ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveCards::Uwp::IAdaptiveActionElement*>* items,
+    std::vector<std::shared_ptr<AdaptiveCards::BaseActionElement>>& containedElements);
+
+HRESULT GenerateSharedImages(
```

Comment:
>GenerateSharedImages [](start = 8, length = 20)

Do you think these specific ones should stay here, or move to the ImageSet, FactSet and ChoiceSet classes?

Reply:
I put them here because the ones that go in the other direction are already here (GenerateFactsProjection, GenerateImagesProjection, etc).  I'm going to leave as is for now unless you feel strongly.

---
In reply to: [149742053](https://github.com/Microsoft/AdaptiveCards/pull/888#discussion_r149742053) [](ancestors = 149742053)

Reply:
I do not feel strongly.

---
In reply to: [149750292](https://github.com/Microsoft/AdaptiveCards/pull/888#discussion_r149750292) [](ancestors = 149750292,149742053)